### DISCUSSION TITLE
vcpkg installation

### DISCUSSION THREAD

**Discussion by matthewclendening** [microsoft/vcpkg#23851] [General]
In the current documentation, vcpkg installation is two steps:
1. Clone repo
2. Run bootstrap script
You can then optionally integrate into msbuild / VS

Recently there is an alternative method described in Marc Goodner’s article [Bootstrap your dev environment with vcpkg artifacts](https://devblogs.microsoft.com/cppblog/vcpkg-artifacts/). There now exists a one step method, like this for a CMD shell:

> curl -LO https://aka.ms/vcpkg-init.cmd && .\vcpkg-init.cmd

This will install vcpkg into the user's “.vcpkg” folder in their profile. However, this is a lightweight install that can’t interact with ports. Given that the official documentation has not changed from the above two steps, I’m assuming this one step method is for supporting the ‘activate’ workflow and environment configuration. I’m wondering if the normal default install experience will be changing and will there be a preferred install location by default?

**Comment by AugP** [comment]
We are still working on bringing together the vcpkg artifacts experience with the experience for ports. There will be a few different ways to get vcpkg on your machine going forward but we want all of them to support both ports and artifacts. Another method will be when we integrate vcpkg into the Visual Studio and Visual Studio Code C++ experiences and ship a prebuilt binary of the tool for that. 

The vcpkg tool may be installed in multiple different places. If installing it from VS, it will end up in your Program Files folder in the VS installation directory. If installing for VS Code, it will likely go where extensions are usually installed. The ports tree will be in a separate non-elevated location though so it can be easily updated. 

If installed independently we leave it up to you to choose. Vcpkg artifacts go in a specific location though you will be able to change that from the default. Vcpkg ports are always installed either in the vcpkg install directory (classic mode) or your project under the vcpkg_installed folder (manifest mode).

**Reply by Osyotr** [reply]
Where will `scripts` folder live and what version will it have?
What about `buildtrees`? Some ports require it to be as little as `C:/b`.
